### PR TITLE
Implements continuous mode

### LINKED ISSUE #19: No continuous Mode
Hello,

I need continuous mode for one of my projet using multiple VL53L0X.

Do you planned to make the support for in your lib ?

Does I need to do the support myself ?

Regards,

### PR DESCRIPTION
**Pull request by Smankusors**
Fixes #19
Also with an example.

Tested it on Nvidia Jetson Nano with 2 VL53L0X (but it's not from Adafruit).

### CODE REVIEW

File: adafruit_vl53l0x.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
   https://github.com/adafruit/circuitpython/releases
 * Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
 """
+import atexit
```
Comment (ladyada) [inline adafruit_vl53l0x.py:48]: atexit isn't always available, please try to make the changes you want without atexit in the library

Comment (Smankusors) [reply adafruit_vl53l0x.py:48]: Ouuu I see, hmm perhaps the continuous mode should be stopped manually from program instead from the library?

Comment (ladyada) [reply adafruit_vl53l0x.py:48]: yep!

File: adafruit_vl53l0x.py [hunk 2]
```diff
@@ -564,6 +587,60 @@ def range(self):
...
+    def continuous_mode(self):
+        """Is the sensor currently in continuous mode?"""
+        return self._continuous_mode
+
+    @continuous_mode.setter
+    def continuous_mode(self, enabled):
+        if enabled:
+            self.start_continuous()
+        else:
+            self.stop_continuous()
```
Comment (tannewt) [inline adafruit_vl53l0x.py:600]: I wonder if we should make this read-only to force use of start_ and stop_. That way there is only one way to do it.

Furthermore, you could make start_continuous return itself and implement`__enter__` and `__exit__` so it is a context manager. That would achieve something similar to `atexit` (which we don't have in CircuitPython.)

Comment (Smankusors) [reply adafruit_vl53l0x.py:600]: hmm interesting... but if we have N VL53L0X sensors, then we will have N `with` statement in the program code though 😨

Comment (tannewt) [reply adafruit_vl53l0x.py:600]: The use of `with` could be optional just in case you want to auto-stop.

Comment (Smankusors) [reply adafruit_vl53l0x.py:600]: hmm actually I changed my program from calling `stopContinuous`, to set all of the XSHUT pins to LOW instead...

**Comment by kattni** [comment]
@Smankusors Please let me know your status on this PR. Let us know if you're stuck or need assistance with anything. Thanks for the contribution!

**Comment by Smankusors** [comment]
> @Smankusors Please let me know your status on this PR. Let us know if you're stuck or need assistance with anything. Thanks for the contribution!

hmm to be honest I'm actually don't want to implement `__enter__` and `__exit__`, kinda confusing. 😵 

Should it create new class to handle the context or just use `self` to handle it? What about multiple sensors? Should it managed by one context manager or managed by itself separately?

thanks for replying btw 😉

**Comment by kattni** [comment]
@tannewt Can you weigh in on this? It's outside my knowledge. Thanks!

**Comment by tannewt** [comment]
> Should it create new class to handle the context or just use `self` to handle it? What about multiple sensors? Should it managed by one context manager or managed by itself separately?

I'd just use self. Multiple sensors could either do nested `with` or do start_continuous and stop_continuous explicitly.

**Comment by Smankusors** [comment]
> I'd just use self. Multiple sensors could either do nested `with` or do start_continuous and stop_continuous explicitly.

Hmm alright. But unfortunately I left my device and sensor on my workplace. And there's still pandemic going on. I will continue to work on this when I got access back.

**Comment by jposada202020** [comment]
Hello @Smankusors have you been able to get access to the device, are you still interested in this PR? let us know. Thanks :)

**Comment by Smankusors** [comment]
> Hello @Smankusors have you been able to get access to the device, are you still interested in this PR? let us know. Thanks :)

uh oh, yeah I do already have access to the device, but then I got busy with other things. I will try to find some free time to continue on this. Thanks for the reminder.

**Comment by Smankusors** [comment]
alright I just implemented the context manager for the continuous mode. Also I add an example for the simple continuous 😉

hmm it seems like the code I wrote kinda look weird, I defined a class (`continuous_manager`) inside a function... is that ok? Or should I move it to somewhere?

**Comment by jposada202020** [comment]
@tannewt Could you review please. Thanks

**Comment by Smankusors** [comment]
> > alright I just implemented the context manager for the continuous mode. Also I add an example for the simple continuous wink
> > hmm it seems like the code I wrote kinda look weird, I defined a class (`continuous_manager`) inside a function... is that ok? Or should I move it to somewhere?
> 
> I think you can simplify it a bit by having `continuous_mode` return `self` and then putting the `__enter__` and `__exit__` functions on the top level class. `__enter__` can call start and `__exit__` would call stop.

I see... how about the failing checks? Should I add the... license? Copy paste from other files?

**Comment by jposada202020** [comment]
@Smankusors for the copyright license yes, that is the way to go, as you adding the example you will need to add your name, or if you prefer you github user.
run Pylint and Black locally on your code. You can do
this manually or using pre-commit. Instructions are available here: https://adafru.it/check-your-code

**Comment by Smankusors** [comment]
OK I simplified the manager, and also add the license to my example codes. Is the license format I entered correct?

**Review by tannewt — APPROVED** [review]
Looks good to me! Thank you!

**Review by FoamyGuy — APPROVED** [review]
tested successfully with feather RP2040 and VL53L0X

**Pull request merged**